mzroll/mainwindow.cpp

```diff
@@ -1639,6 +1656,98 @@ void MainWindow::loadCompoundsFile() {
 	}
 }
 
+// open function for set csv
+void MainWindow::loadSetsFile() {
+	LOGD;
+	QStringList filelist =
+			QFileDialog::getOpenFileNames(this, "Select Set Information File To Load",
+					".",
+					"All Known Formats(*.csv *.tab *.tab.txt);;Tab Delimited(*.tab);;Tab Delimited Text(*.tab.txt);;CSV File(*.csv)");
+
+    if ( filelist.size() == 0 || filelist[0].isEmpty() ) return;
+	if(!loadSetsFile(filelist[0])) {
+		string dbfilename = filelist[0].toStdString();
+		string dbname = mzUtils::cleanFilename(dbfilename);
+
+		QMessageBoxResize msgBox;
+		msgBox.setText(tr("Trouble in loading set information %1").arg(QString::fromStdString(dbname)));
+		msgBox.setIcon(QMessageBoxResize::Warning);
+
+		int ret = msgBox.exec();
+	}
+}
+
+int MainWindow::loadSetsCSVFile(string filename){
+
+    ifstream myfile(filename.c_str());
+    if (! myfile.is_open()) return 0;
+
+    string line;
+    string dbname = mzUtils::cleanFilename(filename);
+    int loadCount=0;
+    int lineCount=0;
+    map<string, int>header;
+    vector<string> headers;
+    static const string allHeadersarr[] = {"sample", "set"};
+    vector<string> allHeaders (allHeadersarr, allHeadersarr + sizeof(allHeadersarr) / sizeof(allHeadersarr[0]) );
+
+    //assume that files are tab delimited, unless matched ".csv", then comma delimited
+    string sep="\t";
+    if(filename.find(".csv") != -1 || filename.find(".CSV") != -1) sep=",";
+    //cerr << filename << " sep=" << sep << endl;
+    while ( getline(myfile,line) ) {
+        if (!line.empty() && line[0] == '#') continue;
+        //trim spaces on the left
+        line.erase(line.find_last_not_of(" \n\r\t")+1);
+        lineCount++;
+
+        vector<string>fields;
+        mzUtils::splitNew(line, sep, fields);
+
+        for(unsigned int i=0; i < fields.size(); i++ ) {
+            int n = fields[i].length();
+            if (n>2 && fields[i][0] == '"' && fields[i][n-1] == '"') {
+                fields[i]= fields[i].substr(1,n-2);
+            }
+            if (n>2 && fields[i][0] == '\'' && fields[i][n-1] == '\'') {
+                fields[i]= fields[i].substr(1,n-2);
+            }
+        }
+
+        if (lineCount==1) {
+            headers = fields;
+            for(unsigned int i=0; i < fields.size(); i++ ) {
+                fields[i] = makeLowerCase(fields[i]);
+                header[fields[i]] = i;
+            }
+            continue;
+        }
+
+        string set;
+		QString sampleName;
+        int N=fields.size();
+
+        if ( header.count("sample")&& header["sample"]<N) 	 sampleName = QString::fromUtf8(fields[ header["sample"] ].c_str());
+        if ( header.count("set")&& header["set"]<N)	set = fields[ header["set"] ];
+		else{
+			set = "NA";
```

Comment:
This creates a new set called NA. The column should be empty. The default should be empty or there should be some condition to check if it is NA then don't process it as setname.